File: javascript/src/models/backgroundTaskType.ts
// this file is @generated

export enum BackgroundTaskType {
  EndpointReplay = "endpoint.replay",
  EndpointRecover = "endpoint.recover",
  ApplicationStats = "application.stats",
  MessageBroadcast = "message.broadcast",
  SdkGenerate = "sdk.generate",
  EventTypeAggregate = "event-type.aggregate",
  ApplicationPurgeContent = "application.purge_content",
  EndpointBulkReplay = "endpoint.bulk_replay",
}

export const BackgroundTaskTypeSerializer = {
  _fromJsonObject(object: any): BackgroundTaskType {
    return object;
  },

  _toJsonObject(self: BackgroundTaskType): any {
    return self;
  },
};

File: kotlin/lib/src/main/kotlin/models/BackgroundTaskType.kt
// This file is @generated
package com.svix.kotlin.models

import com.svix.kotlin.ToQueryParam
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonPrimitive

@Serializable
enum class BackgroundTaskType : ToQueryParam {
    @SerialName("endpoint.replay") ENDPOINT_REPLAY,
    @SerialName("endpoint.recover") ENDPOINT_RECOVER,
    @SerialName("application.stats") APPLICATION_STATS,
    @SerialName("message.broadcast") MESSAGE_BROADCAST,
    @SerialName("sdk.generate") SDK_GENERATE,
    @SerialName("event-type.aggregate") EVENT_TYPE_AGGREGATE,
    @SerialName("application.purge_content") APPLICATION_PURGE_CONTENT,
    @SerialName("endpoint.bulk_replay") ENDPOINT_BULK_REPLAY;

    override fun toQueryParam() = Json.encodeToJsonElement(this).jsonPrimitive.content
}
